addons/website_event/models/event_event.py

```diff
@@ -413,15 +413,16 @@ def _create_menu(self, sequence, name, url, xml_id, menu_type):
         """
         self.browse().check_access('write')
         view_id = False
-        if not url:
+        if not url or (url and xml_id):
             # add_menu=False, ispage=False -> simply create a new ir.ui.view with name
             # and template
             page_result = self.env['website'].sudo().new_page(
                 name=f'{name} {self.name}', template=xml_id,
                 add_menu=False, ispage=False)
             view_id = page_result['view_id']
-            view = self.env["ir.ui.view"].browse(view_id)
-            url = f"/event/{self.env['ir.http']._slug(self)}/page/{view.key.split('.')[-1]}"  # url contains starting "/"
+            if not url:
+                view = self.env["ir.ui.view"].browse(view_id)
+                url = f"/event/{self.env['ir.http']._slug(self)}/page/{view.key.split('.')[-1]}"  # url contains starting "/"
```

Comment:
I don't think we need this nor the change to `_get_website_menu_entries`  anymore, as we don't need a view to store seo data.